File: Treasury.Domain/Model/BudgetSection.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace Treasury.Domain.Model
{
    public partial class BudgetSection
    {
        public BudgetSection()
        {
            BudgetLineItems = new HashSet<BudgetLineItem>();
        }

        public int Id { get; set; }
        public int BId { get; set; }
        public string SectionName { get; set; }
        public DateTime Timestamp { get; set; }

        public virtual Budget BIdNavigation { get; set; }
        public virtual ICollection<BudgetLineItem> BudgetLineItems { get; set; }
    }
}

File: src/main/java/edu/wpi/sga/treasury/domain/model/BudgetSection.java
package edu.wpi.sga.treasury.domain.model;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.time.Instant;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Entity
@Table(name = "BudgetSection")
public class BudgetSection {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID", nullable = false)
    private Integer id;

    @ManyToOne(optional = false)
    @JoinColumn(name = "B_ID", nullable = false)
    private Budget budget;

    @Column(name = "Section_Name", nullable = false)
    private String sectionName;

    @Column(name = "Timestamp", nullable = false)
    private Instant timestamp;

    @OneToMany(mappedBy = "budgetSection")
    private List<BudgetLineItem> budgetLineItems = new ArrayList<>();

}